controllers/managedcluster.go

```diff
@@ -75,34 +108,58 @@ func (r *MultiClusterHubReconciler) ensureKlusterletAddonConfig(m *operatorsv1.M
 		return ctrl.Result{}, err
 	}
 
-	klusterletaddonconfig := getKlusterletAddonConfig()
+	newKlusterletaddonconfig := getKlusterletAddonConfig(m)
```

Comment:
Optional: I think this is clearer
```suggestion
	desiredKlusterletaddonconfig := getKlusterletAddonConfig(m)
```